src/v2/view-builder/internals/BaseView.js

```diff
@@ -19,36 +19,52 @@ export default View.extend({
       '<div class="siw-main-footer"></div>',
 
   initialize () {
+    // Add Views
+    this.add(this.Header, { selector: '.siw-main-header' });
+    this.renderForm();
+    this.add(this.Footer, { selector : '.siw-main-footer' });
+  },
+
+  renderForm () {
+    let subSchemaConfig;
+
+    if (this.form) {
+      this.form.remove();
+      subSchemaConfig = this.form.model.toJSON({verbose: true});
+    }
+
     // Create Model
-    const IonModel = this.createModelClass();
+    const IonModel = this.createModelClass(
+      this.options.currentViewState,
+      subSchemaConfig);
     const model = new IonModel ({
       formName: this.options.currentViewState.name,
     });
 
-    // Add Views
-    this.add(this.Header, { selector: '.siw-main-header' });
-    this.add(this.Body, {
+    if (!subSchemaConfig) {
+      subSchemaConfig = model.toJSON({verbose: true});
+    }
+
+    this.form = this.add(this.Body, {
       selector : '.siw-main-body',
       options: {
         model,
+        subSchemaConfig,
       },
-    });
-    this.add(this.Footer, { selector : '.siw-main-footer' });
-  },
+    }).last();
 
-  postRender () {
-    // If user enterted identifier is not found, API sends back a message with a link to sign up
-    // This is the click handler for that link
-    const appState = this.options.appState;
-    this.$el.find('.js-sign-up').click(function () {
-      appState.trigger('invokeAction', 'select-enroll-profile');
-      return false;
+    _.each(model.attributes, (value, key) => {
+      if (key.match(/sub_schema_local_[^ ]+/)) {
```

Comment:
when using this subschema are the settings like required true preserved? 

if so. When one subschema has value filled and other isnt doenst form submit trigger error?

Reply:
It does but not quite matter because it's `local`.
All sub-uischema are preserving `require: true` hence form validation will kill in.

Reply:
oh yeah they are local.